sphinx-mode.el

```diff
@@ -122,10 +122,32 @@ If BUFFER is not given use the `current-buffer'."
     (find-file (plist-get target :file))
     (goto-char (plist-get target :point))))
 
+(defun sphinx-compile ()
+  "Run 'make' in project root directory and prompt user for target format.
+
+Return absolute path of compiled version of current source file.
+
+To see list of target formats, run 'make help' in a shell."
+  (interactive)
+    (let ((buffer-name-base (file-name-base (buffer-file-name)))
+       (project-root-dir (locate-dominating-file buffer-file-name "Makefile"))
+       (target-format (read-string "Make (default html): " nil nil "html" nil)))
+      (let* ((file-rel-path (file-relative-name buffer-file-name project-root-dir))
+(build-directory (expand-file-name (concat project-root-dir "_build/" target-format "/"))))
+	(shell-command (concat "make -C " project-root-dir " " target-format))
+	(concat build-directory (car (file-name-all-completions (file-name-base file-rel-path) build-directory))))))
```

Comment:
Is the purpose of `file-name-all-completions` to turn `foo.rst` into `foo.html`?  That's pretty clever :)